agave/models/mongo/mongo_model.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, Optional
+
+from cuenca_validations.typing import DictStrAny
+from mongoengine import Document, DoesNotExist, Q
+
+from agave.exc import ObjectDoesNotExist
+from agave.lib.mongoengine.model_helpers import mongo_to_dict
+from agave.models.base import BaseModel
+
+
+class MongoModel(BaseModel, Document):
+    meta = {'allow_inheritance': True}
+
+    def dict(self) -> DictStrAny:
+        return self._dict(mongo_to_dict)
+
+    @classmethod
+    def retrieve(cls, id: str, user_id: Optional[str] = None):
+        try:
+            query = Q(id=id)
+            if user_id:
+                query = query & Q(user_id=user_id)
```

Comment:
this doesn't need to be in the `try` block. It's impossible for these lines to raise `DoesNotExist`

Reply:
Done!